dspace/config/local.cfg.EXAMPLE

```diff
@@ -30,14 +30,18 @@
 # Windows note: Please remember to use forward slashes for all paths (e.g. C:/dspace)
 dspace.dir=/dspace
 
-# DSpace host name - should match base URL.  Do not include port number
-dspace.hostname = localhost
+# URL of DSpace backend ('server' webapp). Include port number etc.
+# This is where REST API and all enabled server modules (OAI-PMH, SWORD, SWORDv2, RDF, etc) will respond
+dspace.server.url = http://localhost:8080/server
 
-# DSpace server backend webapp URL.  Include port number etc.
-dspace.baseUrl = http://localhost:8080/server
+# DSpace UI hostname. Should match dspace.ui.url.  Do not include port number.
```

Comment:
If this should match `dspace.ui.url` then shouldn't we refer to it in `dspace.ui.url` (and `dspace.server.url`)?  That is:
```
dspace.ui.hostname = localhost
dspace.ui.url = http://${dspace.ui.hostname}:3000
dspace.server.url = http://${dspace.ui.hostname}:8080/server
```
(That does make the name `dspace.ui.hostname` confusing, though.)

Reply:
@mwoodiupui : Unfortunately, there's no guarantee that the "hostname" is the *same* for both the frontend (UI) and the backend (REST API).  It might be in many scenarios, but some sites may choose to install these on separate servers.

That's why I chose to rename `dspace.hostname` to `dspace.ui.hostname` to clarify that the expected hostname here is the hostname of the frontend (UI), and *not necessarily* the backend.

I completely agree that this "hostname" is confusing, but I couldn't find a way to easily refactor it out of the codebase altogether.  Perhaps we should build a utility to simply pull the hostname out of the URL rather than configuring it separately.

Reply:
Plucking the hostname out of the URL (which *must* be configured) is better than having a property that has to be coordinated with others, *in code*.  We should do that.

I was thinking of simplifying the *coordination of multiple configuration properties*.  Backend and frontend (and Solr) *can* be on different hosts, but I expect they typically won't be.  In the example we have the same hostname in four places and have to change all four the same way.  However, on reflection I see that there are too many choices, so I suppose my best approach is to give up trying to link it all together.

Reply:
@mwoodiupui : Yes, I think I agree. I'll update this PR to pluck the hostname out of the URL.  Thanks for this feedback!

As for simplifying configuration, I think this will be as simple as we can get.  However, keep in mind, the DSpace configuration system is perfectly OK with you creating new custom configs.  So, locally, anyone could (re)create a `hostname` config and use it to populate both URLs.

Reply:
@mwoodiupui : I've now refactored the code to remove the need for a `hostname` configuration. See the updated PR description above.

I'd appreciate your review here, as I did have to do some code refactoring (especially in OAI-PMH) to ensure the "hostname" could *always* be extracted from the URL.  However, I've added tests for all new utility methods & I've done local tests myself and everything seems to be working.